### DISCUSSION TITLE
Livewire 3 - data not binding issue

### DISCUSSION THREAD

**Discussion by HT-ftw** [livewire/livewire#6208] [1. Help] [answered]
Hi Everyone,

for quite a while I'm struggling with the issue below.
I have this simple component:
```
namespace App\Livewire\Livewiretest;

use Livewire\Component;

class Livewiretest extends Component
{
    public $test='hello';
    
    public function render()
    {
        return view('livewire.livewiretest.livewiretest');
    }
}
```
and 

the following template:
```
<div>

    <input type="text" wire:model="test">

    <p>this is a: {{ $test }}</p>
</div>
```

In my understanding when I write something in the input the $test variable should be updated and should display however it is not working and nothing is happening.

If I change the code like this:
```
namespace App\Livewire\Livewiretest;

use Livewire\Component;

class Livewiretest extends Component
{
    public $test='hello';
    
    public function render()
    {
        return view('livewire.livewiretest.livewiretest');
    }

    public function save()
    {
        
    }
```
```
<div>
    {{-- Stop trying to control. --}}

    <form>
    <input type="text" wire:model="test">
    <button wire:click.prevent="save()">Submit</button>
    </form>

    <p>this is a: {{ $test }}</p>
</div>
```

and this works after submitting the $test variable is updated.

So can anybody help me and explain what I am doing wrong in the first case?:)
Thanks
HT

**Comment by ildyria** [comment, marked as answer]
It is because it is debounced. You need to add `.live` to your wire:model. In other words:
```html
<input type="text" wire:model.live="test">
```

See here: https://livewire.laravel.com/docs/upgrading#wiremodel

**Reply by HT-ftw** [reply]
Thank you it helped a lot, I should have read the docs more carefully.
Still got one more question should the also work with a <select>?
